src/dotnet/HQ.Server/Services/HolidayServiceV1.cs

```diff
@@ -205,5 +210,71 @@ public async Task<Result<ImportHolidayV1.Response>> ImportHolidayV1(ImportHolida
             Updated = updated
         };
     }
+    public async Task BackgroundAutoGenerateHolidayTimeEntryV1(CancellationToken ct)
+    {
+        var today = DateOnly.FromDateTime(DateTime.UtcNow);
+        var startDate = today.GetPeriodStartDate(Period.Week);
+        var endDate = today.GetPeriodEndDate(Period.Week);
+
+        var holidays = _context.Holidays
+        .AsNoTracking()
+        .AsQueryable();
+        var jurisdicitons = Enum.GetValues(typeof(Jurisdiciton));
+        var holidayChargeCode = await _context.ChargeCodes.Where(t => t.Project!.Name.ToLower().Contains("holiday")).FirstOrDefaultAsync(ct);
+        if (holidayChargeCode == null)
+        {
+            return;
+        }
+        foreach (Jurisdiciton jurisdiciton in jurisdicitons)
+        {
+            var upcomingHolidays = await holidays.Where(t => t.Date >= startDate && t.Date <= endDate && t.Jurisdiciton == jurisdiciton).ToListAsync(ct);
+            foreach (var upcomingHoliday in upcomingHolidays)
+            {
+
+                if (upcomingHoliday == null)
+                {
+                    return;
+                }
+                var staff = _context.Staff.
+                AsNoTracking()
+                .AsQueryable();
+
+
+                staff = staff.Where(t => t.EndDate == null && t.Jurisdiciton == jurisdiciton);
+                var times = _context.Times
+                .AsNoTracking()
+                .AsQueryable()
+                .Include(t => t.Staff);
+                var staffWithHoliday = await times.Where(t => t.Date == upcomingHoliday.Date && t.ChargeCode == holidayChargeCode && t.Staff.Jurisdiciton == jurisdiciton && t.Staff.EndDate == null).Select(t => t.Staff).ToListAsync(ct);
```

Comment:
Instead of comparing the charge code object itself, compare using `t.ChargeCodeId == holidayChargeCode.Id`. With `AsNoTracking()` I'm not sure the equality operator on the object will work correctly.